src/librbd/ExclusiveLock.cc

```diff
@@ -528,44 +470,6 @@ void ExclusiveLock<I>::handle_reacquire_lock(int r) {
   ldout(cct, 10) << this << " " << __func__ << ": r=" << r << dendl;
 
   assert(m_state == STATE_REACQUIRING);
-  if (r < 0) {
-    if (r == -EOPNOTSUPP) {
-      ldout(cct, 10) << this << " " << __func__ << ": "
-                     << "updating lock is not supported" << dendl;
-    } else {
-      lderr(cct) << this << " " << __func__ << ": "
-                 << "failed to update lock cookie: " << cpp_strerror(r)
-                 << dendl;
-    }
-
-    if (!is_shutdown()) {
-      // queue a release and re-acquire of the lock since cookie cannot
-      // be updated on older OSDs
-      execute_action(ACTION_RELEASE_LOCK, nullptr);
-
-      assert(!m_actions_contexts.empty());
-      ActionContexts &action_contexts(m_actions_contexts.front());
-
-      // reacquire completes when the request lock completes
-      Contexts contexts;
-      std::swap(contexts, action_contexts.second);
-      if (contexts.empty()) {
-        execute_action(ACTION_REQUEST_LOCK, nullptr);
-      } else {
-        for (auto ctx : contexts) {
-          ctx = new FunctionContext([ctx, r](int acquire_ret_val) {
-              if (acquire_ret_val >= 0) {
-                acquire_ret_val = r;
-              }
-              ctx->complete(acquire_ret_val);
-            });
-          execute_action(ACTION_REQUEST_LOCK, ctx);
-        }
-      }
-    }
-  } else {
-    m_cookie = m_new_cookie;
-  }
 
   complete_active_action(STATE_LOCKED, 0);
```

Comment:
If the re-acquire fails, you'd still need to execute the release / reacquire steps above